skimage/feature/scattering.py

```diff
@@ -0,0 +1,287 @@
+        he images will be cropped.
+    wavelet_filters : Dictionary with the multiresolution wavelet filter bank
+        Dictionary of vectors obtained after calling:
+            >>>> px = 32 #number of pixels of the images
+            >>>> J = np.log2(px) #number of scales
+            >>>> wavelet_filters = multiresolution_filter_bank_morlet2d(px, J=J)
+    m : int
+        Order of the scattering transform, which can be 0, 1 or 2.
+    Returns
+    -------
+    S : 4D array_like
+        Scattering transform of the x signals, of size
+        (N, num_coeffs, spatial_coefs, spatial_coefs). For more information
+        see _[1] _[2]
+    U : array_like
+        Result before applying the lowpass filter and subsampling.
+    S_tree : dictionary
+        Dictionary that allows to access the scattering coefficients (S)
+        according to the layer and indices. More specifically:
+            Zero-order layer: The only available key is 0:
+            S_tree[0] : returns all the coefficients of the 0-order scattering
+             transform
+
+            First-order layer: The keys are tuples with (j, l) indexing
+            S_tree[(j, l)] : returns the first order coefficients for scale 'j'
+             and angle 'l'
+            S_tree[((j1, l1), (j2, l2))] : the second order for scale 'j1',
+            angle 'l1' on the first layer, and 'j2', 'l2' in the second layer.
+
+        The number of coefficients for each entry is
+        (N, spatial_coefs, spatial_coefs)
+    Raises
+    ------
+    UserWarning
+        If the size of x is not (N, px, px) and informs that the images with be
+        cropped to (N, px, px)
+    UserWarning
+        If no wavelet filters, the function creates a multiresolution set of
+        filters with predefined settings, but warns about the parameters and
+        suggests precomputing the filters.
+    ValueError
+        If the value of m is not 0, 1, or 2.
+    References
+    ----------
+    .. [1] Bruna, J., Mallat, S. 'Invariant Scattering Convolutional Networks'.
+    IEEE Transactions on PAMI, 2012.
+    .. [2] Oyallon, E. et Mallat, S. 'Deep Roto-translation Scattering for
+    Object Classification'. CVPR 2015.
+    Examples
+    --------
+    Processing a set of 3 (randomly generated) images:
+       >>>> import numpy as np
+       >>>> from scattering.filter_bank \
+                import multiresolution_filter_bank_morlet2d
+       >>>> from scattering.scattering import scattering
+       >>>> px = 32 #number of pixels of the images
+       >>>> images = range(0,3*px*px).reshape(3,px,px) #create images
+       >>>> wavelet_filters,lp = \
+                multiresolution_filter_bank_morlet2d(px,J=np.log2(px))
+       >>>> S,U = scattering(images,m=2)
+    """
+    # Check that the input data is correct:
+    # 1.- Signals are squared, otherwise, crop
+    assert_nD(x, 3, 'x')  # check that the images are correctly stacked
+    num_signals, px, py = x.shape
+    if px != py:
```

Comment:
Because it can become cumbersome the definition of the filters so that they have a good covering of the spectral domain (good littlewood paley). Normally, we define squared filters and then change the images, but since this changes a lot the geometry of the image, we decided that it was better to make the user very aware of this, thus forcing the user to do it before she/he calls the scattering function.